test/intl402/Segmenter/constructor/constructor/options-granularity-word.js

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-Intl.Segmenter
+description: Checks handling of valid values for the granularity option "word" when string text includes URLs, email addresses, #, Dates and acronyms
+features: [Intl.Segmenter]
+---*/
+
+const testValuesByLength = [
+  ["F.C.B", 1],
+  ["F.C.BALL", 1],
+  ["F.C. BALL", 4],
+  ["Please P.T.O 123 times. 123.3242432", 10],
+  ["at 12:43 PM", 7],
+  ["in 21/03/2013.", 8],
+  ["https://example.com/tc39/test262", 9],
+  ["https://example.com/tc39/test262?id=test", 13],
```

Comment:
Unsure about the rules for URL segmentation and could be interpreted differently